unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp

```diff
@@ -0,0 +1,250 @@
+
+    Kokkos::fence();
+
+    /// for comparison send it to host
+    auto a_host = Kokkos::create_mirror_view(acopy);
+    auto x_host = Kokkos::create_mirror_view(x);
+    auto b_host = Kokkos::create_mirror_view(b);
+    auto w_host = Kokkos::create_mirror_view(w);
+
+    Kokkos::deep_copy(a_host, acopy);    
+    Kokkos::deep_copy(x_host, x);
+    Kokkos::deep_copy(b_host, b);
+
+    /// this is least square; we cannot expect high
+    typedef typename ats::mag_type mag_type;
+    const mag_type eps = 1e3 * ats::epsilon();
+
+    for (int k=0;k<N;++k) {
+      mag_type residual(0), norm(0);
+
+      for (int l=0;l<2;++l) {
+	for (int i=0;i<BlkSize;++i) {
+	  value_type tmp(0);
+	  for (int j=0;j<BlkSize;++j) {	
+	    tmp += a_host(k,i,j)*x_host(k,j,l);
+	  }
+	  w_host(k,i) = tmp - b_host(k,i,l);
+	}
+	for (int i=0;i<BlkSize;++i) {
+	  value_type tmp(0);
+	  for (int j=0;j<BlkSize;++j) {	
+	    tmp += a_host(k,j,i)*w_host(k,j);
+	  }
+	  residual += ats::abs(tmp)*ats::abs(tmp);
+	  norm += ats::abs(b_host(k,i,l))*ats::abs(b_host(k,i,l));	
+	}
+      }
+      //printf("norm %e, residual %e, rel res %e\n", norm, residual, residual/norm);
+      EXPECT_NEAR_KK( residual/norm, mag_type(0), eps);
+    }
+  }
+}
+
+
+template<typename DeviceType,
+         typename ValueType,
+	 typename IntType,
+         typename AlgoTagType>
+int test_batched_solve_utv2() {
+#if defined(KOKKOSKERNELS_INST_LAYOUTLEFT)
+  {
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutLeft, DeviceType> MatrixViewType;
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutLeft, DeviceType> VectorViewType;
+    typedef Kokkos::View<IntType**,   Kokkos::LayoutLeft, DeviceType> PivViewType;
+    typedef Kokkos::View<ValueType**, Kokkos::LayoutRight,DeviceType> WorkViewType;
+    Test::impl_test_batched_solve_utv2<DeviceType,MatrixViewType,VectorViewType,
+				       PivViewType,WorkViewType,AlgoTagType>(     0, 10);
+    for (int i=1;i<10;++i) {                                                                                        
+      //printf("Testing: LayoutLeft,  Blksize %d\n", i); 
+      Test::impl_test_batched_solve_utv2<DeviceType,MatrixViewType,VectorViewType,
+					 PivViewType,WorkViewType,AlgoTagType>(1024, i);
+    }
+  }
+#endif
+#if defined(KOKKOSKERNELS_INST_LAYOUTRIGHT)
+  {
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutRight,DeviceType> MatrixViewType;
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutRight,DeviceType> VectorViewType;
+    typedef Kokkos::View<IntType**,   Kokkos::LayoutRight,DeviceType> PivViewType;
+    typedef Kokkos::View<ValueType**, Kokkos::LayoutRight,DeviceType> WorkViewType;
+    Test::impl_test_batched_solve_utv2<DeviceType,MatrixViewType,VectorViewType,
+				       PivViewType,WorkViewType,AlgoTagType>(     0, 10);
+    for (int i=1;i<10;++i) {                                                                                        
+      //printf("Testing: LayoutRight, Blksize %d\n", i); 
```

Comment:
Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.